app/layout/layout.phtml

```diff
@@ -66,5 +66,18 @@
 	<span class="msg"><?= $msg ?></span>
 	<a class="close" href=""><?= _i('close') ?></a>
 </div>
+
+<!-- Popup-->
+<div id="popup">
+	<div id="popup-content">
+		<div id="popup-close" class="popup-row">&times;</div>
+		<div id="popup-txt" class="popup-row"></div>
+		<div id="popup-iframe-container" class="popup-row">
+			<div id="popup-iframe-sub">
+				<iframe id="popup-iframe" frameborder="0" sandbox="allow-scripts"></iframe>
```

Comment:
Maybe we need to use the new security header
https://scotthelme.co.uk/a-new-security-header-feature-policy/

Reply:
Just by curiosity, why do you prefer not to use iframe ?

So yes, it was for simplicity + I thought it would be better to don't interact with current page style. I was not sure of what was covered by `sanitizeHTML`, btw.

And it's why I created a double mechanism for the Popup: one to show content by iframe, and one to show content by changing a simple "p" node (`<div id="popup-txt" class="popup-row"></div>`), so the pop-up is a bit more flexible. Not sure it's the right way to do it, but…

Reply:
> I thought it would be better to don't interact with current page style.

Wait, that was a conscious choice? I thought of it as a for now perfectly functional (and possibly for years to come) but I'd just style it with the existing .flux_content .content classes.

Reply:
The simplicity was an important point in my choice too ;p

But I guess it would simplify the javascript problem. And I see that `XMLHttpRequest` is already used in `init_slider_observers`, so I guess I can guet inspiration from there...

Reply:
We can probably use fetch now btw.

https://caniuse.com/#feat=fetch

Reply:
So, I tried to remove the iframe and use this weird `fetch` API to put the result in the innerHTML of a div.

For some reason, it seems to works… (JavaScript is a language which doesn't make any sense to me, I don't know how all of you are able to understand what you are doing with it!).

But I feel that the result is worst than with an iframe, in fact. I have tons of errors because of CSP (not able to load remote images, not able to load inline images, not able to load movies, etc.)

The problem is because this popup content (i.e. part of the article) now "run" in the CSP of the "subscription" controller (i.e. default CSP) which is very restricive. I'm not sure we want to open it more just for the preview...

The iframe had the advantage to better control the request / result context.
<img width="503" alt="Screenshot 2020-02-01 at 19 47 37" src="https://user-images.githubusercontent.com/1971659/73597380-bb822400-452b-11ea-8ff7-025a5a10095a.png">

Reply:
Ftr, I don't think there's anything wrong with iframes in this context but I'm open to arguments to the contrary. But not "interacting" with the page style struck me as a bug, not a feature. :-)